outlook/calendar/pkg/commands/searchEvents.go

```diff
@@ -21,19 +22,30 @@ func SearchEvents(ctx context.Context, query string, start, end time.Time) error
 		return fmt.Errorf("failed to create client: %w", err)
 	}
 
-	result, err := graph.SearchEvents(ctx, c, query, start, end)
+	calendars, err := graph.ListCalendars(ctx, c)
 	if err != nil {
-		return fmt.Errorf("failed to search events: %w", err)
+		return fmt.Errorf("failed to list calendars: %w", err)
 	}
 
-	calendarEvents := map[graph.CalendarInfo][]models.Eventable{}
-	for cal, events := range result {
-		if len(events) > 0 {
-			calendarEvents[cal] = events
+	calendarEventsInSubject := map[graph.CalendarInfo][]models.Eventable{}
+	calendarEventsInPreview := map[graph.CalendarInfo][]models.Eventable{}
```

Comment:
nit: better to over allocate for bigger requests.
```suggestion
	calendarEventsInSubject := make(map[graph.CalendarInfo][]models.Eventable, len(calendars))
	calendarEventsInPreview := make(map[graph.CalendarInfo][]models.Eventable, len(calendars))
```